### PR TITLE
rust: initial version of rust stack

### PR DESCRIPTION
**Pull request by skoh7645**
### Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stack-structure.md).

### Modifying an existing stack:

- [ ] Updated the stack version in `stack.yaml`

Initial version of rust stack

### Contributing a new stack:

- Describe how application dependencies are managed:

- Explain how Appsody file watcher is utilized:

- Describe other Appsody environment variables defined by the stack image:

- Describe any limitations and known issues:

### Related Issues:
Related to #356

### CODE REVIEW

**Comment by EnriqueL8** [comment]
For the `appsody debug` command, there is no direct command in Rust such as in `npm`. Rust uses `lldb`, so we are investigating into remote debugging inside the docker container using `lldb`. We were able to run the executable from a remote shel using the lldb command line, but we would need to link the lldb connection to an IDE for better user experience. This would need looking into LSPs for Rust. https://github.com/rust-lang/rls

**Comment by Kamran64** [comment]
Tested and checked (tick off as completed):

- [ ] ```appsody run```
- [ ] ```appsody test```
- [ ] ```appsody debug```
- [ ] ```appsody build```
- [ ] ```appsody deploy```
    - [ ] Using Knative
    - [ ] Using operator deploy
- [ ] Build script (Lint, list etc.)
- [ ] README structure
- [ ] Dockerfile-stack structure

File: experimental/rust/image/Dockerfile-stack [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM rust:1.37
+
+ENV APPSODY_MOUNTS=.:/project/user-app
+ENV APPSODY_DEPS=$CARGO_HOME/registry
+
+ENV APPSODY_WATCH_DIR="/project/user-app"
+ENV APPSODY_WATCH_IGNORE_DIR=
```
Comment (neeraj-laad) [inline experimental/rust/image/Dockerfile-stack]: We can delete this line

File: experimental/rust/image/Dockerfile-stack [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ENV APPSODY_WATCH_REGEX="(^.*.rs$)"
+
+ENV APPSODY_RUN="cargo run"
+ENV APPSODY_RUN_ON_CHANGE="cargo run"
+ENV APPSODY_RUN_KILL=false
+
+ENV APPSODY_TEST="cargo test"
+ENV APPSODY_TEST_ON_CHANGE=""
+ENV APPSODY_TEST_KILL=false
```
Comment (neeraj-laad) [inline experimental/rust/image/Dockerfile-stack]: can probably delete these too

File: experimental/rust/image/project/filename.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+bin_name=$(awk '$0 == "[[bin]]" {i=1;next};i && i++ <= 1' Cargo.toml)
+
+echo "$bin_name" | sed -e 's/name = "\(.*\)"/\1/'
```
Comment (neeraj-laad) [inline experimental/rust/image/project/filename.sh]: This looks like something that should be within the stack not in the project template

File: experimental/rust/image/project/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+RUN chmod +x filename.sh \
+ && chown rust:rust $(./filename.sh)
+
+USER rust
+
+ENV PORT 8000
+
+EXPOSE 8000
+
+CMD ["bash", "-c", "./$(./filename.sh)"]
```
Comment (neeraj-laad) [inline experimental/rust/image/project/Dockerfile]: Could we get the name of the app in stage 1 and then use that info in stage 2 as in get rid of the filename.sh in the final image?

Comment (skoh7645) [reply experimental/rust/image/project/Dockerfile]: I have deleted `filename.sh`- we now get the name of the generated binary solely in the `dockerfile`

**Pull request merged**